be/src/formats/orc/apache-orc/c++/src/Reader.cc

```diff
@@ -1060,10 +1066,34 @@ void RowReaderImpl::lazyLoadNext(ColumnVectorBatch& data, uint64_t numValues) {
     } else {
         reader->lazyLoadNext(data, numValues, nullptr);
     }
+    lazyLoadRowInStripe += numValues;
+    lazyLoadLastUsedRowInStripe = lazyLoadRowInStripe;
 }
 
 void RowReaderImpl::lazyLoadSkip(uint64_t numValues) {
-    reader->lazyLoadSkip(numValues);
+    uint64_t toRow = lazyLoadRowInStripe + numValues;
+    // we have two options to locate `toRow`:
+    // 1. use seek to row group, and skip rest rows(# = toRow % rowindexstripe)
+    // 2. or skip rest rows directly(# = roRow - lazyLoadLastUsedRowInStripe)
+    // we can assume `seek to row group` costs skipping X rows
+    // then #1 cost is X + toRow % rowindexstripe
+    // #2 cost is (row - lazyLoadLastUsedRowInStripe)
+    const uint64_t ROW_INDEX_STRIDE = footer->rowindexstride();
+    static const uint64_t SEEK_TO_ROW_GROUP_COST = 4096;
```

Comment:
How do we get this `SEEK_TO_ROW_GROUP_COST` value?

Reply:
# of row index stride is usually 10000, so I take half of it as the estimated cost of doing seek row group operation.